### DISCUSSION TITLE
How to implement semantic-release in existing projects?

### DISCUSSION THREAD

**Discussion by angelicaflausino** [semantic-release/semantic-release#2864] [Q&A] [answered]
Hi everyone, I have an existing project and we are looking into implementing semantic-release,

Project context
- Angular Library
- Our commits don't follow angular message conventions.
- Our versioning model we use the major version based on the angular version and at the moment we have 2 versions: 
       - 8.0.0 (angular 8) 
       - 15.0.0 (angular 15) 

Based on this context I have some doubts...

1. How to adjust existing commits to angular's message convention?
2. How to keep major version based on angular version?

Thanks if anyone has any similar experience in this context and can share some tips with me.

**Comment by travi** [comment]
> 2\. How to keep major version based on angular version?

syncing your versions with an external source is, by definition, not following semantic versioning for your packages. this use case is, therefore,  not supported by semantic-release

**Reply by angelicaflausino** [reply]
OK. But could I generate git tags 8.0.0 and 15.0.0 and be able to use semantic-release from there?

**Reply by MarcLoupias** [reply, marked as answer]
Yes, semantic-release search for the last tag found on the trunk, just push manually the 15.0.0 tag on your trunk and starts using conventional-commit from there. The tool will calculate the new tag version from 15.0.0 with commits msg since 15.0.0. Have done it before, it works.

Best way to test it is to do it in a test repo ;)

**Reply by angelicaflausino** [reply]
@MarcLoupias thanks for reply, I created the tags and it worked!
